src/Paprika/Store/AbandonedList.cs

```diff
@@ -166,15 +158,13 @@ private void Register(DbAddress head, IBatchContext batch)
     {
         if (MaxCount == EntriesCount)
         {
-            // No place, attach it to the youngest batch id from the sorted BatchIds
+            // No place, attach it to the last page which has the youngest batch id. Addresses
+            // are always sorted by batch ids, hence the last page.
             var maxAt = MaxCount - 1;
-            Debug.Assert(MaxCount > 2 && BatchIds[MaxCount - 1] > BatchIds[MaxCount - 2]);
```

Comment:
Removed the debug assert to keep the code shorter. We can fetch the corresponding batch ids from the pages, but not sure if it is required?

Reply:
I think it's ok now.